### PR TITLE
Test framework: add support to mock GetVersion

### PR DESCRIPTION
**Pull request by yiminc**
adding support to mock GetVersion in unit test.

### CODE REVIEW

File: internal/workflow_testsuite.go [hunk 1]
```diff
@@ -306,6 +307,17 @@ func (t *TestWorkflowEnvironment) OnRequestCancelExternalWorkflow(domainName, wo
 	return t.wrapCall(call)
 }
 
+// OnGetVersion setup a mock for workflow.GetVersion() call. By default, if mock is not setup, the GetVersion call from
+// workflow code will always return the maxSupported version. Make it not possible to test old version branch. With this
+// mock support, it is possible to test code branch for different versions.
+// Note: if mock for GetVersion is setup, the test framework expect all calls to GetVersion from the tested workflow are
+// all mocked. It is not supported to have one GetVersion call mocked and the second GetVersion call not mocked. You need
+// to either not mock any GetVersion call, or mock all GetVersion calls.
```
Comment (mfateev) [inline internal/workflow_testsuite.go]: I don't think it is acceptable. The problem is that the workflow might reference some library that calls GetVersion but the implementer of the workflow might not even know about it. I believe it should be possible to mock each "changeId" individually.

File: internal/internal_workflow_testsuite.go [hunk 2]
```diff
@@ -1622,7 +1622,31 @@ func (env *testWorkflowEnvironmentImpl) SideEffect(f func() ([]byte, error), cal
...
+		if mockFn := m.getMockFn(mockRet); mockFn != nil {
+			executor := &activityExecutor{name: mockMethodForGetVersion, fn: mockFn}
+			reflectValues := executor.executeWithActualArgsWithoutParseResult(nil, args)
+			if len(reflectValues) != 1 || !reflect.TypeOf(reflectValues[0].Interface()).AssignableTo(reflect.TypeOf(retVerion)) {
+				panic(fmt.Sprintf("mock of GetVersion has incorrect return type, expected workflow.Version, but actual is %T (%v)",
+					reflectValues[0].Interface(), reflectValues[0].Interface()))
+			}
+			return reflectValues[0].Interface().(Version)
+		} else {
+			if len(mockRet) != 1 || !reflect.TypeOf(mockRet[0]).AssignableTo(reflect.TypeOf(retVerion)) {
```
Comment (dmetzgar) [inline internal/internal_workflow_testsuite.go]: Could this be `reflect.TypeOf((*Version)(nil))` instead?

Also, fix spelling of `retVerion`.

File: internal/workflow_testsuite.go (hunk 1 above)
Comment (yiminc) [reply internal/workflow_testsuite.go]: good point, let me update

File: internal/internal_workflow_testsuite.go (hunk 2 above)
Comment (yiminc) [reply internal/internal_workflow_testsuite.go]: remove retVersion and use DefaultVersion instead.

File: internal/workflow_testsuite.go [hunk 3]
```diff
@@ -310,11 +310,11 @@ func (t *TestWorkflowEnvironment) OnRequestCancelExternalWorkflow(domainName, wo
 // OnGetVersion setup a mock for workflow.GetVersion() call. By default, if mock is not setup, the GetVersion call from
 // workflow code will always return the maxSupported version. Make it not possible to test old version branch. With this
 // mock support, it is possible to test code branch for different versions.
-// Note: if mock for GetVersion is setup, the test framework expect all calls to GetVersion from the tested workflow are
-// all mocked. It is not supported to have one GetVersion call mocked and the second GetVersion call not mocked. You need
-// to either not mock any GetVersion call, or mock all GetVersion calls.
+//
+// Note: mock can be setup for a specific changeID. Or if mock.Anything is used as changeID then all calls to GetVersion
+// will be mocked.
```
Comment (mfateev) [inline internal/workflow_testsuite.go]: You can add that mock for a specific changeID has priority over Anything mock.

**Pull request merged**